.github/workflows/tics-charm.yaml

```diff
@@ -0,0 +1,41 @@
+name: TICS
+
+on:
+  workflow_call:
+    secrets:
+      TICSAUTHTOKEN:
+        required: true
+
+
+jobs:
+  analyze:
+    runs-on: [self-hosted, linux, amd64, tiobe, jammy]
+
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4
+
+      - name: Install system dependencies
+        run: sudo apt-get update && sudo apt-get install -y python3.10-venv pipx
+
+      - name: Install tox using pipx
+        run: |
+          pipx install tox
+          pipx inject tox tox-uv
+
+      - name: Run tox tests to create coverage.xml
+        run: tox run -e unit
+
+      - name: move results to necessary folder for TICS
+        run: |
+          mkdir cover
+          mv coverage.xml cover/coverage.xml
+
+      - name: Run TICS analysis with github-action
+        uses: tiobe/tics-github-action@88cb795a736d2ca885753bec6ed2c8b03e3f892f # v3
+        with:
+          mode: qserver
+          project: vault-operator
```

Comment:
We probably shouldn't hardcode the project right?

Reply:
Ugh, I looked through this last night to see if I left anything like this and somehow missed it. Thanks for the fresh eyes.